### PR TITLE
docs: api/grn_obj: move grn_obj_is_locked() reference to the header file

### PR DESCRIPTION
**Pull request by otegami**
GitHub: GH-1817

Prepare to switch to documents automatically generated by Doxygen.

### CODE REVIEW

File: include/groonga/groonga.h [hunk 1]
```diff
@@ -1343,6 +1343,15 @@ grn_obj_unlock(grn_ctx *ctx, grn_obj *obj, grn_id id);
  */
 GRN_API grn_rc
 grn_obj_clear_lock(grn_ctx *ctx, grn_obj *obj);
+/**
+ * \brief Check if an object is currently locked.
+ *
+ * \param ctx The context object.
+ * \param obj The target object whose lock status is to be checked.
+ *
+ * \return 0 if there are no locks. Return a non-zero unsigned integer
```
Comment (abetomo) [inline include/groonga/groonga.h:1352]:
```suggestion
 * \return `0` if there are no locks. Return a non-zero unsigned integer
```

数値は ` で囲っているので合わせると良いです。

Comment (otegami) [reply include/groonga/groonga.h]: fix: [de705a6](https://github.com/groonga/groonga/pull/1995/commits/de705a6ceb5dcef00b35fff32901b6914bb122a8)

**Review by abetomo — APPROVED** [review]

File: include/groonga/groonga.h [hunk 2]
```diff
@@ -1343,6 +1343,15 @@ grn_obj_unlock(grn_ctx *ctx, grn_obj *obj, grn_id id);
  */
 GRN_API grn_rc
 grn_obj_clear_lock(grn_ctx *ctx, grn_obj *obj);
+/**
+ * \brief Check if an object is currently locked.
+ *
+ * \param ctx The context object.
+ * \param obj The target object whose lock status is to be checked.
+ *
+ * \return `0` if there are no locks. Return a non-zero unsigned integer
```
Comment (kou) [inline include/groonga/groonga.h:1352]:
```suggestion
 * \return `0` if there are no locks. Return the number of acquired locks
```

**Comment by otegami** [comment]
Thank you for reviewing. I've just fixed what you pointed out.

**Pull request merged**